consumer/dvo_rules_consumer_test.go

```diff
@@ -0,0 +1,300 @@
+		MessageProcessor: consumer.DVORulesProcessor{},
+	}
+}
+
+func dummyDVOConsumer(s storage.DVORecommendationsStorage, allowlist bool) consumer.Consumer {
+	brokerCfg := broker.Configuration{
+		Address: "localhost:1234",
+		Topic:   "topic",
+		Group:   "group",
+	}
+	if allowlist {
+		brokerCfg.OrgAllowlist = mapset.NewSetWith(types.OrgID(1))
+		brokerCfg.OrgAllowlistEnabled = true
+	} else {
+		brokerCfg.OrgAllowlistEnabled = false
+	}
+	return createDVOConsumer(brokerCfg, s)
+}
+
+func TestDVORulesConsumer_New(t *testing.T) {
+	helpers.RunTestWithTimeout(t, func(t testing.TB) {
+		sarama.Logger = log.New(os.Stdout, saramaLogPrefix, log.LstdFlags)
+
+		mockStorage, closer := ira_helpers.MustGetMockStorage(t, true)
+		defer closer()
+
+		mockBroker := sarama.NewMockBroker(t, 0)
+		defer mockBroker.Close()
+
+		mockBroker.SetHandlerByMap(ira_helpers.GetHandlersMapForMockConsumer(t, mockBroker, testTopicName))
+
+		mockConsumer, err := consumer.NewDVORulesConsumer(broker.Configuration{
+			Address: mockBroker.Addr(),
+			Topic:   testTopicName,
+			Enabled: true,
+		}, mockStorage)
+		helpers.FailOnError(t, err)
+
+		err = mockConsumer.Close()
+		helpers.FailOnError(t, err)
+	}, testCaseTimeLimit)
+}
+
+func TestDeserializeEmptyDVOMessage(t *testing.T) {
+	c := consumer.KafkaConsumer{MessageProcessor: consumer.DVORulesProcessor{}}
+	_, err := consumer.DeserializeMessage(&c, []byte(""))
+	assert.EqualError(t, err, "unexpected end of JSON input")
+}
+
+func TestDeserializeDVOMessageWithWrongContent(t *testing.T) {
+	const message = `{"this":"is", "not":"expected content"}`
+	c := consumer.KafkaConsumer{MessageProcessor: consumer.DVORulesProcessor{}}
+	_, err := consumer.DeserializeMessage(&c, []byte(message))
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "missing required attribute")
+}
+
+func TestDeserializeDVOMessageWithImproperJSON(t *testing.T) {
+	const message = `"this_is_not_json_dude"`
+	c := consumer.KafkaConsumer{MessageProcessor: consumer.DVORulesProcessor{}}
+	_, err := consumer.DeserializeMessage(&c, []byte(message))
+	assert.EqualError(
+		t,
+		err,
+		"json: cannot unmarshal string into Go value of type consumer.incomingMessage",
+	)
+}
+
+func TestDeserializeDVOMessageWithImproperMetrics(t *testing.T) {
+	consumerMessage := `{
+		"OrgID": ` + fmt.Sprint(testdata.OrgID) + `,
+		"ClusterName": "` + string(testdata.ClusterName) + `",
+		"LastChecked": "` + testdata.LastCheckedAt.Format(time.RFC3339) + `",
+		"Metrics": {
+			"we_dont_know_yet": "what_format_it_will_have",
+            "but": "this should be deserialized properly"
+		}
+	}`
+	c := consumer.KafkaConsumer{MessageProcessor: consumer.DVORulesProcessor{}}
+	message, err := consumer.DeserializeMessage(&c, []byte(consumerMessage))
+	helpers.FailOnError(t, err)
+	assert.Equal(t, types.OrgID(1), *message.Organization)
+	assert.Equal(t, testdata.ClusterName, *message.ClusterName)
+}
+
+//TODO: We don't know yet what a proper message is
```

Comment:
shall we mark the PR as draft?

Reply:
no no, these are things that depend on the rules team. I left the test there without implementing on purpose, so it's clear that there are parts of this code that are not here because we cannot implement them as of today.